modvendor/download/report.go

```diff
@@ -109,6 +109,41 @@ func (r Report) Verbose() string {
 	return strings.Join(report, "\n")
 }
 
+// RemoveIgnoredByKind removes all ignored from this report that have errors
+// with the given kind.
+func (r *Report) RemoveIgnoredByKind(kind errors.Kind) {
+	ignored := []IgnoredVendor{}
+	for _, v := range r.Ignored {
+		if !errors.IsKind(v.Reason, kind) {
+			ignored = append(ignored, v)
+		}
+	}
+	r.Ignored = ignored
+}
+
+// IsEmpty returns true if the report is empty (nothing to report).
+func (r Report) IsEmpty() bool {
+	return len(r.Vendored) == 0 &&
+		len(r.Ignored) == 0 &&
+		r.Error == nil
+}
+
+func (r *Report) merge(other Report) {
+	if other.Error != nil {
```

Comment:
can't it be simplified using errors.List? like:
```
errs := errors.L(r.Error, other.Error)
r.Error = errs.AsError()
```
?
because the errors.L() and errs.Append() ignores nil errors.

Reply:
or in one line:
```
r.Error = errors.L(r.Error, other.Error).AsError()
```
I didn't test but I think this (should) work.

Reply:
I think it can, yeah we do filter out nil automatically, cool :100: